content/learn/beginner/control-flow-1.md

```diff
@@ -0,0 +1,181 @@
+
+{{< callout context="caution" title="Comparing across Types" icon="outline/alert-triangle" >}}
+
+Just like you cannot quite compare apples and oranges, you cannot compare values of different types. For instance, you
+cannot compare a number to a string. The bot will throw an error if you try to do so, you will have to convert either of
+them to the other type first.
+
+{{< /callout >}}
+
+We provide the following comparison functions in custom commands:
+
+- `eq` (equals `==`)
+- `ne` (not equals `!=`)
+- `lt` (less than `<`)
+- `le` (less than or equal to `<=`)
+- `gt` (greater than `>`)
+- `ge` (greater than or equal to `>=`)
+
+These functions can only compare basic data types as introduced in [Data Types 1](/learn/beginner/datatypes-1). Strings
+are compared on a byte-by-byte basis. Please refer to the [functions documentation](/docs/reference/templates/functions)
+for the syntax of these functions.
+
+## Blocks And Scope
+
+In [Data Types 1](/learn/beginner/datatypes-1), we introduced the concept of variables. Variables are used to store
+values and give them a name. In programming, variables have a *scope*, which defines where in the code the variable can
+be accessed. Think of each scope as a "container" for things inside it.
+
+Often, you will want to have a variable available across multiple scopes. In custom commands, the variable is accessible
+in the scope in which it was defined, as well as all nested scopes within. Let us assume that we want to assign a
+coolness value, which should be true if the user's name is "alice". We can achieve this by defining a variable in
+the outer scope and re-assigning, using the `=` operator, its value in the inner scope:
+
+```go
+{{ $isCool := false }}
+
+{{ if eq .User.Username "alice" }}
+  {{ $isCool = true }}
+{{ end }}
+
+Are you cool? {{ $isCool }}
+```
+
+It is considered good practice to define variables in the smallest scope possible. This makes your code easier to read
+and understand, as you do not have to search through the entire codebase to find where a variable was defined.
+
+{{< callout context="caution" title="Definition and Reassignment" icon="outline/alert-triangle" >}}
+
+In custom commands, you use `:=` to define a variable, and `=` to reassign a variable. The bot will not throw an error
+if you try to re-define a variable using `:=`, but it will not affect the outer scoped variable.
+
+{{< /callout >}}
+
+<div style="float: right; margin-left: 7px;">
+
+![Responses: template: :XX: unexpected EOF](unexpected-eof.png)
+
+</div>
+
+Try and recall the last example from the introduction of if-statements and notice how, despite us having multiple blocks
+of branching code, there's still only one `{{ end }}` action.
+
+Each control structure's start (`if`, `with`, etc.) must eventually have such a closing statement. Not doing so will
+result in an error similar to `template: :XX: unexpected EOF`, where `:XX:` is the line number where the error
+occurred---an example shown above.
```

Comment:
If we're moving the image below this block of text in the raw markdown, it will flow into the next section.

![image](https://github.com/user-attachments/assets/a159da28-9416-42ec-a516-e7a1dc621a05)

Having it in the centre and below the text doesn't feel nice to me, as it is relatively small—in the rendered page it is above and to the right of this sentence.

Reply:
How about beside the paragraph that references it, rather than the first paragraph? It feels slightly out of place there to me.
![2024-07-22T122832](https://github.com/user-attachments/assets/d27c7734-fc2e-404c-bb48-21ce5ff4697d)